x-pack/plugins/monitoring/server/lib/apm/get_apm_info.js

```diff
@@ -0,0 +1,139 @@
+    uuid: apmUuid,
+    transportAddress: get(stats, 'beat.host', null),
+    version: get(stats, 'beat.version', null),
+    name: get(stats, 'beat.name', null),
+    type: capitalize(get(stats, 'beat.type')) || null,
+    output: capitalize(get(stats, 'metrics.libbeat.output.type')) || null,
+    configReloads: get(stats, 'metrics.libbeat.config.reloads', null),
+    uptime: get(stats, 'metrics.beat.info.uptime.ms', null),
+    eventsTotal: getDiffCalculation(eventsTotalLast, eventsTotalFirst),
+    eventsEmitted: getDiffCalculation(eventsEmittedLast, eventsEmittedFirst),
+    eventsDropped: getDiffCalculation(eventsDroppedLast, eventsDroppedFirst),
+    bytesWritten: getDiffCalculation(bytesWrittenLast, bytesWrittenFirst),
+  };
+}
+
+export async function getApmInfo(req, apmIndexPattern, { clusterUuid, apmUuid, start, end }) {
+  checkParam(apmIndexPattern, 'apmIndexPattern in beats/getBeatSummary');
+
+  const filters = [
+    { term: { 'beats_stats.beat.uuid': apmUuid } },
+    { term: { 'beats_stats.beat.type': 'apm-server' } }
+  ];
+  const params = {
+    index: apmIndexPattern,
+    size: 1,
+    ignoreUnavailable: true,
+    filterPath: [
+      'hits.hits._source.beats_stats.beat.host',
+      'hits.hits._source.beats_stats.beat.version',
+      'hits.hits._source.beats_stats.beat.name',
+      'hits.hits._source.beats_stats.beat.type',
+      'hits.hits._source.beats_stats.metrics.libbeat.output.type',
+      'hits.hits._source.beats_stats.metrics.libbeat.pipeline.events.published',
+      'hits.hits._source.beats_stats.metrics.libbeat.pipeline.events.total',
+      'hits.hits._source.beats_stats.metrics.libbeat.pipeline.events.dropped',
+      'hits.hits._source.beats_stats.metrics.libbeat.output.write.bytes',
+      'hits.hits._source.beats_stats.metrics.libbeat.config.reloads',
+      'hits.hits._source.beats_stats.metrics.beat.info.uptime.ms',
+      'hits.hits.inner_hits.first_hit.hits.hits._source.beats_stats.metrics.libbeat.pipeline.events.published',
+      'hits.hits.inner_hits.first_hit.hits.hits._source.beats_stats.metrics.libbeat.pipeline.events.total',
+      'hits.hits.inner_hits.first_hit.hits.hits._source.beats_stats.metrics.libbeat.pipeline.events.dropped',
+      'hits.hits.inner_hits.first_hit.hits.hits._source.beats_stats.metrics.libbeat.output.write.bytes',
+    ],
+    body: {
+      sort: { timestamp: { order: 'desc' } },
+      query: createQuery({
+        start,
+        end,
+        clusterUuid,
+        metric: ApmMetric.getMetricFields(),
+        filters
+      }),
+      collapse: {
+        field: 'beats_stats.metrics.beat.info.ephemeral_id', // collapse on ephemeral_id to handle restart
+        inner_hits: {
+          name: 'first_hit',
+          size: 1,
+          sort: { 'beats_stats.timestamp': 'asc' }
+        }
+      }
+    }
+  };
+
+  const { callWithRequest } = req.server.plugins.elasticsearch.getCluster('monitoring');
+
+  // Fetch additional information
+  const params2 = {
```

Comment:
What contains params2 and what is just in params? Perhaps we can find a more descriptive var name?

Reply:
Absolutely. I'd like to find a way to combine these into a single request assuming we end up sticking with this data in the UI